tests/apidoc_login_test.go

```diff
@@ -0,0 +1,42 @@
+package tests
+
+import (
+	"encoding/json"
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/tidwall/gjson"
+	"testing"
+)
+
+func TestLogin(t *testing.T) {
+	deployment := Deploy(t, "login", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	t.Run("parallel", func(t *testing.T) {
+		// sytest: POST /login can log in as a user
+		t.Run("POST /login can login as user", func(t *testing.T) {
+			t.Parallel()
+			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, json.RawMessage(`{
+				"auth": {
+					"type": "m.login.password"
+				},
+				"username": "post-login-user",
+				"password": "superuser"
+			}`))
+			res = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+				"auth": {
+					"type": "m.login.dummy"
+				},
+				"username": "post-login-user",
+				"password": "superuser"
```

Comment:
I don't think this structure is quite correct. The [matching SyTest](https://github.com/matrix-org/sytest/blob/develop/tests/10apidoc/02login.pl#L70-L101) sets:

```perl
type     => "m.login.password",
identifier => {
   type => "m.id.user",
   user => $user_id,
},
password => $password,
```

Reply:
Hi @callahad I have corrected the order and have even captured the user_id from the register response. But how can I pass that userId variable in the json raw message. I would really appreciate any help here 😕

Reply:
Great, thank you!

> how can I pass that userId variable in the json raw message

In this case, hardcoding it to "post-login-user" is probably fine. In general, the easiest way is probably to break up the string and use `+` to concatenate things ([playground](https://play.golang.org/p/9pwWDdn_0Hi)). For example:

```go
`{"user": "` + userid + `"}`
```

You could also use `fmt.Sprintf` ([playground](https://play.golang.org/p/ST5YpRg3GG0)) to do something like:

```go
fmt.Sprintf(`{"user": "%s"}`, userid)
```

But again, for now, just hardcode it 😄 

All that said, I think the JSON structure is still a little off. Instead of sending JSON like:

```json
{
  "auth": {
    "type": "m.login.password"
  },
  "identifier": {
    "type": "m.id.user",
    "user": "post-login-user"
  },
  "username": "post-login-user",
  "password": "superuser"
}
```

I think you only need to send:

```json
{
  "type": "m.login.password",
  "identifier": {
    "type": "m.id.user",
    "user": "post-login-user",
  },
  "password": "superuser"
}
```

(Meaning: I think you can drop the `auth` and `username` keys altogether.)

Reply:
I tried using `f.Sprintf` but since the complete Raw Message parameter is under `` literal, the string concatenation was not working as expected. 
But as you said, I will hardcode it for now, thanks 😄